File: 🟢 Easy/989 - Add to Array-Form of Integer/_989AddtoArray-FormofInteger.js
/**
 * @param {number[]} num
 * @param {number} k
 * @return {number[]}
 */
var addToArrayForm = function (num, k) {
    let res = [];
    let i = num.length - 1;

    while (i >= 0 || k > 0) {
        if (i >= 0) {
            k += num[i];
        }
        res.push(k % 10);
        k = Math.floor(k / 10);
        i--;
    }

    res.reverse();
    return res;
};

File: 🟢 Easy/989 - Add to Array-Form of Integer/_989AddtoArray-FormofInteger.cpp
class Solution {
public:
    vector<int> addToArrayForm(vector<int>& num, int k) {
        vector<int> res;
        int i = num.size() - 1;
        while (i >= 0 || k > 0) {
            if (i >= 0) {
                k += num[i];
            }
            res.push_back(k % 10);
            k /= 10;
            i--;
        }
        reverse(res.begin(), res.end());
        return res;
    }
};